### DISCUSSION TITLE
Mend & Renovate & Naming plans

### DISCUSSION THREAD

**Discussion by rarkins** [renovatebot/renovate#29885] [Suggest an Idea]
### Tell us more.

Here's a heads-up about naming adjustments around Renovate, and I'm also happy to hear feedback.

Today we have multiple variants and ways to run Renovate, and it can be confusing. I'll try to list them all in the most descriptive/distinguishable way here:

- The Renovate Open Source CLI. Built from this repo, published primarily as the npmjs package `renovate`. This is also available as:
	* Docker images, full and normal (slim) modes, published to GHCR and Docker Hub
	* GitHub Action
	* GitLab Pipeline
- Renovate Community Edition (CE). A free but closed source wrapper around the Renovate CLI, requires registration for license key from Mend.io.
- Renovate Enterprise Edition (EE). A commercial closed source wrapper around Renovate CLI, like CE but allowing for horizontal scalability, Merge Confidence APIs
- Mend Renovate App for GitHub.com. A free hosted service of Renovate, currently 100% free
- Mend App for Bitbucket Cloud. A hosted app by Mend. Has both a free plan (Renovate-only) as well as Mend's full AppSec solution (SCA, SAST, etc).

A related challenge has been when to use the Mend name or not (Mend Renovate or simply Renovate), whether to use "Bot" or not (Renovate Bot or Renovate), etc. In general we've dropped the "bot" part but it's totally ok to use it whenever it needs clarification that we're talking Renovate the dependency updater and not renovating bathrooms. For a while I experimented with using "Mend Renovate" when it's Mend-hosted or Mend-enhanced Renovate such as CE/EE, and "Renovate" when it's the pure Open Source part. It's worked OK, but I think this approach created confusion.

Going forward I've decided to align on this approach:
- The term "Renovate" can still be used to refer to Renovate in general, where the type of Renovate is not particularly important.
- All Renovate will be officially be "Mend Renovate" to keep naming simpler, similar to Microsoft Windows, Adobe Photoshop, etc.
- We don't use "Bot" in official names, but you're welcome to say "Renovate Bot"
- Each Renovate product/variant will have a name, so they can be differentiated, and we'll document them all in one place

Therefore the planned list of terms/names is:

**Mend Renovate CLI**: "CLI" refers mostly to this repo's output - the original/core/OSS Renovate. Available as npm package, Docker images, GitHub Action, GitLab pipeline template, etc. We'll keep the simple `renovate` npm package name, but migrate the rest under Mend's orgs.

**Mend Renovate Community**: "Community" will refer to any non-OSS but free offering from Mend, e.g. the existing Mend Renovate Community Edition, or the existing free Renovate plan on the Mend Renovate app for GitHub.com and Bitbucket Cloud.

**Mend Renovate Enterprise**: "Enterprise" will refer to any commercial/non-free Renovate offering from Mend, e.g. the existing Mend Renovate Enterprise Edition, or any future paid plan on the Mend hosted apps.

**Comment by MindTooth** [comment]
I usually refer to Renovate Bot as the "individual" that interacts with us. 🤖

**Reply by adam-moss** [reply]
I refer to it as "our most productive employee" 🤣

**Comment by HonkingGoose** [comment]
## Names feedback

I like the new names. I agree that we should standardize. We have so many ways to run Renovate, it's easy to get lost.

### CLI

> **Mend Renovate CLI**: "CLI" refers mostly to this repo's output - the original/core/OSS Renovate. Available as npm package, Docker images, GitHub Action, GitLab pipeline template, etc. We'll keep the simple `renovate` npm package name, but migrate the rest under Mend's orgs.

I like how the CLI name leads you towards the concept of a "program you run".

### Community

> **Mend Renovate Community**: "Community" will refer to any non-OSS but free offering from Mend, e.g. the existing Mend Renovate Community Edition, or the existing free Renovate plan on the Mend Renovate app for GitHub.com and Bitbucket Cloud.

A possible confusion is that Community implies "for the common good" or "for the commons", so maybe people expect the community stuff to be open source. But this is a really minor nitpick.

People will understand that Community means: "thing you're getting from Mend, for free, for the community".

### Enterprise

> **Mend Renovate Enterprise**: "Enterprise" will refer to any commercial/non-free Renovate offering from Mend, e.g. the existing Mend Renovate Enterprise Edition, or any future paid plan on the Mend hosted apps.

With software, Enterprise has always meant that you need to pay to use it. So Enterprise is good for any paid offerings.

## Draft list of todos for rename

It will be a lot of work to update the docs, readmes, and code. Here's a draft todo list. I'll let you make proper issues for these todos, when you're ready to start work:

- [ ] Update the [Style guide](https://github.com/renovatebot/renovate/blob/main/docs/development/style-guide.md):
  - [ ] Create new chapter: _Names for Renovate variants_
  - [ ] Describe when/how to use the names
  - [ ] Describe if/when to use the `bot` modifier to denote actions from the bot (Individual/bot that interacts with the user)
- [ ] Update the Renovate docs (`docs/usage`), and the docs's homepage (`index.md`) to use the correct name for each variant
- [ ] Update the readme in the repository
- [ ] Update the [Renovate docs repository's `includes/abbreviations.md` file](https://github.com/renovatebot/renovatebot.github.io/blob/main/includes/abbreviations.md), as this currently shows a hover text for the strings: `the Mend Renovate App` and `The Mend Renovate App`
- [ ] Update user-facing text in things like:
  - [ ] Dependency Dashboard
  - [ ] Config Migration PRs
  - [ ] Renovate "normal" PRs
  - [ ] Renovate's onboarding PR
  - [ ] Mend Renovate Developer Portal

**Comment by jamietanna** [comment]
Sounds like a good plan to try and simplify the naming, and make it a bit easier for newer folks to get up and running on what's what.

One thing I will note is that - in a transition period at $work - we've got folks using `Mend Renovate App for GitHub.com` (which we refer to as `(hosted) Mend Renovate app`), and are moving towards `Mend Renovate Community Edition (CE)` (which we refer to as `(self-hosted) Renovate` or `Mend Renovate Community Edition (CE)`)

I wonder if it's worth making a distinction between hosted and self hosted Mend Renovate Community?

**Reply by rarkins** [reply]
We are tentatively considering the terms "Server" for self-hosted and "Cloud" for Mend-hosted. "Self-hosted" and "Mend-hosted" are more strictly correct, but also less suitable to be added to a product name. Cloud/Server are the terms which GitHub uses for GitHub Enterprise and it seems to not cause too much confusion.

**Reply by HonkingGoose** [reply]
# Summary

> We are tentatively considering the terms "Server" for self-hosted and "Cloud" for Mend-hosted. "Self-hosted" and "Mend-hosted" are more strictly correct, but also less suitable to be added to a product name. Cloud/Server are the terms which GitHub uses for GitHub Enterprise and it seems to not cause too much confusion.

Cloud/Server are common names, people understand what it means.

A product name must be easy to understand. So we can afford to be _mostly_ technically correct. "Mend-hosted Renovate" is technically correct, but lacks the punch of "Renovate Server".

We should explain that "(Mend) Renovate Server" means _the user's server_. So the homepage/docs/readme should have a line like: "(Mend) Renovate Server, host on your own hardware".

### Names for platform hosted

| Platform name | Hosted name      | Notes                              |
| :------------ | :--------------- | :--------------------------------- |
| Azure         | DevOps           |                                    |
| Bitbucket     | Cloud            |                                    |
| GitHub        | Enterprise Cloud |                                    |
| GitLab        | GitLab.com       |                                    |
| GitLab        | Dedicated        | Single-tenant Sofware as a Service |

### Names for self-hosting

| Platform name | Self-hosted name  | Notes                                          |
| :------------ | :---------------- | :--------------------------------------------- |
| Azure         | DevOps Server     |                                                |
| Bitbucket     | Data Center       |                                                |
| Bitbucket     | Server            | Support for Server ended on Feb. 15, 2024 [^1] |
| GitHub        | Enterprise Server |                                                |
| GitLab        | Self Managed      |                                                |

#### Bitbucket Data Center

Data Center sounds like I'll need a room full of servers to run the thing. In comparison running a Server seems easy... :upside_down_face: 

#### GitLab Self Managed

Self managed is a good shorthand for: _you_ must maintain and update _your_ stuff.

So "Renovate Self Managed" could work, but "Renovate Server" still sounds best.

[^1]: [Atlassian, end of life notice for Server products](https://www.atlassian.com/migration/assess/journey-to-cloud)

**Comment by adam-moss** [comment]
From a naming perspective I think they're all just Renovate (Mend Renovate if you prefer, I don't think it necessary but then my usage pre-dates it being whitesource too).

In my opinion what you're really talking about is the license/pricing tier, e.g.

Free - open source cli or container images; self-hosted
Starter - free with extra features like job scheduler and and github app; self or cloud hosted 
Pro - $ with features X, Y; self or cloud hosted 
Enterprise - $$$ with features X, Y, and Z; self or cloud hosted

**Comment by MPV** [comment]
We've started seeing `Renovate plan: Legacy` showing up: 
><img width="400" alt="Skärmavbild 2025-01-10 kl  11 59 21" src="https://github.com/user-attachments/assets/390ba0d0-636a-424e-a425-0c7791cb091e" />

So I started wondering what this `Legacy` plan is, and went back to this issue to refresh my memory, but "legacy" is something I didn't see in this discussion.

Do you have any updated news to share on where this is heading?
